cfg/windows.cfg

```diff
@@ -4045,6 +4052,80 @@ HFONT CreateFont(
       <not-bool/>
     </arg>
   </function>
+  <!--HANDLE WINAPI CreateMutex(
+  _In_opt_ LPSECURITY_ATTRIBUTES lpMutexAttributes,
+  _In_     BOOL                  bInitialOwner,
+  _In_opt_ LPCTSTR               lpName);-->
+  <function name="CreateMutex,CreateMutexA,CreateMutexW">
+    <returnValue type="HANDLE"/>
+    <noreturn>false</noreturn>
+    <arg nr="1">
+      <not-uninit/>
+      <not-bool/>
+    </arg>
+    <arg nr="2">
+      <not-uninit/>
+    </arg>
+    <arg nr="3">
+      <not-uninit/>
+      <not-bool/>
+    </arg>
+  </function>
+  <!--HANDLE WINAPI CreateMutexEx(
+  _In_opt_ LPSECURITY_ATTRIBUTES lpMutexAttributes,
+  _In_opt_ LPCTSTR               lpName,
+  _In_     DWORD                 dwFlags,
+  _In_     DWORD                 dwDesiredAccess);-->
+  <function name="CreateMutexEx,CreateMutexExA,CreateMutexExW">
+    <returnValue type="HANDLE"/>
+    <noreturn>false</noreturn>
+    <arg nr="1">
+      <not-uninit/>
+      <not-bool/>
+    </arg>
+    <arg nr="2">
+      <not-uninit/>
+      <not-bool/>
+    </arg>
+    <arg nr="3">
+      <not-uninit/>
+      <valid>0,1</valid>
+    </arg>
+    <arg nr="4">
+      <not-uninit/>
+    </arg>
+  </function>
+  <!--HANDLE WINAPI OpenMutex(
+  _In_ DWORD   dwDesiredAccess,
+  _In_ BOOL    bInheritHandle,
+  _In_ LPCTSTR lpName);-->
+  <function name="OpenMutex,OpenMutexA,OpenMutexW">
+    <returnValue type="HANDLE"/>
+    <noreturn>false</noreturn>
+    <arg nr="1">
+      <not-uninit/>
+    </arg>
+    <arg nr="2">
+      <not-uninit/>
+    </arg>
+    <arg nr="3">
+      <not-uninit/>
+      <not-null/>
+      <not-bool/>
+      <strz/>
+    </arg>
+  </function>
+  <!--BOOL WINAPI ReleaseMutex(
+  _In_ HANDLE hMutex);-->
+  <function name="ReleaseMutex">
+    <returnValue type="BOOL"/>
+    <noreturn>false</noreturn>
+    <leak-ignore/>
```

Comment:
Isn't this function necessary to avoid a leak?

Reply:
No. ReleaseMutex only unblocks a Mutex that has been blocked before by a call to CreateMutex* with an according argument or by a wait* function. Forgetting ReleaseMutex would result in a deadlock which is fatal, but i see no way (yet or soon) to statically check if it has been forgotten.
There is only a leak if the HANDLE returned by CreateMutex* or OpenMutex is not closed via CloseHandle.